lib/Doctrine/DBAL/Driver/AbstractSQLAnywhereDriver.php

```diff
@@ -46,6 +46,8 @@
     public function convertException($message, DriverException $exception)
     {
         switch ($exception->getErrorCode()) {
+            case '-306':
```

Comment:
There are even more I suppose:

http://dcx.sap.com/1200/en/saerrors/errm307.html
http://dcx.sap.com/1200/en/saerrors/errm684.html
http://dcx.sap.com/1200/en/saerrors/errm210.html
http://dcx.sap.com/1200/en/saerrors/errm1175.html
http://dcx.sap.com/1200/en/saerrors/errm1281.html

Not sure about this one:
http://dcx.sap.com/1200/en/saerrors/errm1344.html